test/spec/templates.spec.js

```diff
@@ -1,180 +1,183 @@
-var chai = require('chai')
+'use strict';
+
+var _ = require('lodash')
+  , chai = require('chai')
   , expect = chai.expect
-  , sinon = require('sinon')
-  , sinonChai = require('sinon-chai')
-  , Promise = require('../../lib/Promise');
+  , sinon = require('sinon');
+
+require('sinon-as-promised');
 
-chai.use(sinonChai);
+chai.use(require('sinon-chai'));
+chai.use(require('chai-as-promised'));
 
 describe('Templates Library', function() {
   var client, templates;
 
   beforeEach(function() {
     client = {
-      get: sinon.stub().returns(Promise.resolve({})),
-      post: sinon.stub().returns(Promise.resolve({})),
-      put: sinon.stub().returns(Promise.resolve({})),
-      delete: sinon.stub().returns(Promise.resolve({}))
+      get: sinon.stub().resolves({}),
+      post: sinon.stub().resolves({}),
+      put: sinon.stub().resolves({}),
+      delete: sinon.stub().resolves({})
     };
 
     templates = require('../../lib/templates')(client);
   });
 
-  describe('all Method', function() {
-    it('should call client get method with the appropriate uri', function(done) {
-      templates.all(function(err, data) {
-        expect(client.get.firstCall.args[0].uri).to.equal('templates');
-        done();
-      });
+  describe('list Method', function() {
```

Comment:
i guess we should have a test cases of passing back errors generated by  `client.get`. as per current test, if we, in future, handle errors differently, all tests will still pass.